### PR TITLE
Fix array return type

### PR DESCRIPTION
**Pull request by GiggleLiu**
* avoid returning subarrays,
* add tests for +, -, = and isapprox

### CODE REVIEW

**Comment by GiggleLiu** [comment]
I have to merge quickly because I need to tag a new version for tomorrow's community call.
Feel free to comment back. @Roger-luo

File: src/PermMatrix.jl [hunk 1]
```diff
@@ -44,6 +44,9 @@ function PermMatrix(
     PermMatrix{Tv,Ti,Vv,Vi}(perm, vals)
 end
 
+Base.:(==)(d1::PermMatrix, d2::PermMatrix) = SparseMatrixCSC(d1) == SparseMatrixCSC(d2)
```
Comment (Roger-luo) [inline src/PermMatrix.jl:47]: Why can't you compare the value directly? Why covert it to CSC?

File: src/PermMatrix.jl [hunk 2]
```diff
@@ -44,6 +44,9 @@ function PermMatrix(
...
+Base.isapprox(d1::PermMatrix, d2::PermMatrix; kwargs...) = isapprox(SparseMatrixCSC(d1), SparseMatrixCSC(d2); kwargs...)
```
Comment (Roger-luo) [inline src/PermMatrix.jl:48]: Same as above

File: src/PermMatrix.jl (hunk 1 above)
Comment (GiggleLiu) [reply src/PermMatrix.jl:47]: If there are two entries being zero, then their order are not important. The logic is too complicated.

**Pull request merged**